bin/auth-api/src/routes/auth.routes.ts

```diff
@@ -13,29 +14,41 @@ import {
 } from "../services/auth.service";
 import { setCache, getCache } from "../lib/cache";
 import {
-  createOrUpdateUserFromAuth0Details, getUserByEmail,
+  createOrUpdateUserFromAuth0Details,
+  getUserByEmail,
 } from "../services/users.service";
 import { validate } from "../lib/validation-helpers";
 
-import {
-  userRoleForWorkspace,
-} from "../services/workspaces.service";
+import { userRoleForWorkspace } from "../services/workspaces.service";
 import { router } from ".";
 
+const FALLBACK_REDIR_PORT = 9003;
+
+const parseCliRedirectPort = (port: string[] | string | undefined): number => {
+  if (typeof port !== "string") {
+    return FALLBACK_REDIR_PORT;
+  }
+
+  const portNumber = Number.parseInt(port);
+  if (isNaN(portNumber) || portNumber < 1) {
+    return FALLBACK_REDIR_PORT;
+  }
+
+  return portNumber;
+};
+
 router.get("/auth/login", async (ctx) => {
-  // TODO: can read from querystring info about where request originated
-  // so that we can shoot later directly back to the right place, skipping auth portal
+  // passing in cli_redir=PORT_NO will begin the auth flow for the si cli
+  const cliRedirect = parseCliRedirectPort(ctx.request.query.cli_redir);
```

Comment:
The only question I have here is what happens when a user logs in from the webapp - we are doing to return 9003 - that's not going to affect the web app? I am sure you have tested that scenario of course!